chain/network/src/peer_manager/peer_store/mod.rs

```diff
@@ -129,9 +132,18 @@ impl Inner {
         } else {
             // If doesn't have the address attached it is not verified and we add it
             // only if it is unknown to us.
-            self.peer_states
-                .entry(peer_info.id.clone())
-                .or_insert_with(|| KnownPeerState::new(peer_info, clock.now_utc()));
+            if !self.peer_states.contains(&peer_info.id) {
+                if let Some((_, popped_peer_state)) = self
+                    .peer_states
+                    .push(peer_info.id.clone(), KnownPeerState::new(peer_info, clock.now_utc()))
+                {
+                    // If a peer was evicted from peer_states due to the bounded cache size
+                    // and it has an address, remove the corresponding entry from addr_peers
+                    if let Some(popped_peer_addr) = popped_peer_state.peer_info.addr {
+                        self.addr_peers.remove(&popped_peer_addr);
```

Comment:
FYI, I don't think we want this information at all. For sure it is not useful for establishing new connections (since addr is missing).